src/handle_action.cpp

```diff
@@ -1469,7 +1469,7 @@ static void cast_spell()
 static bool assign_spellcasting( Character &you, spell &sp, bool fake_spell )
 {
     if( you.is_armed() && !sp.has_flag( spell_flag::NO_HANDS ) &&
-        !you.weapon.has_flag( flag_MAGIC_FOCUS ) ) {
+        !you.weapon.has_flag( flag_MAGIC_FOCUS ) && sp.check_if_component_in_hand( you ) ) {
```

Comment:
Wouldn't this need to be negated for check_if_component_in_hand?

Reply:
The function itself handles it, at the very least when i debug played it, it worked

Reply:
Thinking about it again now that you pointed it out, i might have forgotten to add the negation

Reply:
What I'm troubled by is that it currently would, if all the other conditions were met, reject casting if the character has a component in hand... I would think it would be `&& !sp.check_if_component_in_hand( you )`.

Reply:
Sorry about that, added the negation back now.

Looking back in my commits, somewhere between these astyle commits i removed && !sp.check_if_component_in_hand( you ) and forgot to add it back in src/handle_action, now it shoul be there.